test/ExternalTest.ts

```diff
@@ -0,0 +1,161 @@
+import {assert} from "chai";
+import * as https from "https";
+import {VirtualAlexa} from "../src/core/VirtualAlexa";
+
+describe("Test address API mocks", function() {
+    it("Calls address API for full address", (done) => {
+        const myFunction = function(event: any, context: any) {
+            callAddressAPI(event.context.System.apiAccessToken, event.context.System.device.deviceId)
+                .then((response: any) => {
+                assert.equal(response.payload.addressLine1, "address line 1");
+                assert.equal(response.statusCode, 200);
+                done();
+            });
+        };
+
+        const virtualAlexa = VirtualAlexa.Builder()
+            .handler(myFunction)
+            .sampleUtterancesFile("./test/resources/SampleUtterances.txt")
+            .intentSchemaFile("./test/resources/IntentSchema.json")
+            .create();
+
+        virtualAlexa.addressAPI().returnsFullAddress({
+            addressLine1: "address line 1",
+            addressLine2: "address line 2",
+            addressLine3: "address line 3",
+            city: "city",
+            countryCode: "country code",
+            districtOrCounty: "district",
+            postalCode: "postal",
+            stateOrRegion: "state",
+        });
+
+        virtualAlexa.launch().then();
```

Comment:
adding .then()  is doing nothing. The internal async code is still going to be executed on the .launch(). A good alternative would be to have  

virtualAlexa.launch()
.then(() => done())
.catch(error => done(error))

And remove the done from the internal handler